test/metabase/upload_test.clj

```diff
@@ -1160,111 +1188,178 @@
+                            upload/current-database    (constantly (mt/db))]
+                (let [table (create-upload-table!
+                             {:col->upload-type (ordered-map/ordered-map
+                                                 :_mb_row_id      ::upload/auto-incrementing-int-pk
+                                                 :biginteger      ::upload/int
+                                                 :float           ::upload/float
+                                                 :text            ::upload/varchar-255
+                                                 :boolean         ::upload/boolean
+                                                 :date            ::upload/date
+                                                 :datetime        ::upload/datetime
+                                                 :offset_datetime ::upload/offset-datetime)
+                              :rows [[1000000,1.0,"some_text",false,"2020-01-01","2020-01-01T00:00:00","2020-01-01T00:00:00+00:00"]]})
+                      csv-rows ["biginteger,float,text,boolean,date,datetime,offset_datetime"
+                                "2000000,2.0,some_text,true,2020-02-02,2020-02-02T02:02:02,2020-02-02T02:02:02+02:00"]
+                      file  (csv-file-with csv-rows (mt/random-name))]
+                  (is (some? (append-csv! {:file     file
+                                           :table-id (:id table)})))
+                  (testing "Check the data was uploaded into the table correctly"
+                    (is (= [[1 1000000 1.0 "some_text" false "2020-01-01T00:00:00Z" "2020-01-01T00:00:00Z" "2020-01-01T00:00:00Z"]
+                            [2 2000000 2.0 "some_text" true "2020-02-02T00:00:00Z" "2020-02-02T02:02:02Z" "2020-02-02T00:02:02Z"]]
+                           (rows-for-table table))))
+                  (io/delete-file file))))))))))
 
 (deftest append-no-rows-test
   (mt/test-driver :h2
     (with-uploads-allowed
       (testing "Append should succeed with a CSV with only the header"
-        (doseq [csv-rows [["id,name"]]]
-          (mt/dataset (basic-db-definition)
-            (let [table (t2/select-one :model/Table :db_id (mt/id))
-                  _     (t2/update! :model/Table (:id table) {:is_upload true})
+        (let [csv-rows ["name"]]
+          (mt/with-empty-db
+            (let [table (create-upload-table!)
                   file  (csv-file-with csv-rows (mt/random-name))]
               (is (= {:row-count 0}
-                     (upload/append-csv! {:file     file
-                                          :table-id (:id table)})))
+                     (append-csv! {:file     file
+                                   :table-id (:id table)})))
               (testing "Check the data was not uploaded into the table"
                 (is (= [[1 "Obi-Wan Kenobi"]]
                        (rows-for-table table))))
               (io/delete-file file))))))))
 
+(deftest append-mb-row-id-1-test
```

Comment:
```suggestion
(deftest append-mb-row-id-1-test
```

I'm not a fan of naming tests with numbers, I think we can come up with a short and meaningful name here. it helps reader scan quickly what has been covered.